packages/core/src/index.ts

```diff
@@ -220,6 +237,67 @@ export default class Engine<Name extends string = string> {
 		return this.publicParsedRules
 	}
 
+	getSituation(): Situation<Name> {
+		return this.publicSituation
+	}
+
+	safeGetSituation({
+		situation,
+		shouldThrowError = false,
+	}: {
+		situation: Situation<Name>
+		shouldThrowError?: boolean
+	}): Situation<Name> {
+		const situationCopy = JSON.parse(JSON.stringify(situation))
```

Comment:
Le deepclone n'est pas nécessaire, et même dangeureux : on peut avoir des expressions déjà parsée dans la situation, cela peut donc être lourd. Un clone simple devrait suffire : 
```suggestion
		const situationCopy = {...situation}

Reply:
(@bjlaa je pense qu'on ne gère pas bien les situations parsées dans le script de migration de publicodes/tools d'ailleurs)

Reply:
Okay intéressant ! Les situations parsées sont celles sur lesquelles chaque clé a été évaluée ?